overlays/nixpkgs.nix

```diff
@@ -1,7 +1,61 @@
 # Copyright 2024 Edgeless Systems GmbH
 # SPDX-License-Identifier: AGPL-3.0-only
 
-final: prev: {
+final: prev:
+# TODO(miampf): Remove unneccessary block once https://github.com/NixOS/nixpkgs/pull/345326 is merged into unstable nixpkgs
+let
+  # Builder for Azure CLI extensions. Extensions are Python wheels that
+  # outside of nix would be fetched by the CLI itself from various sources.
+  mkAzExtension =
+    {
+      pname,
+      url,
+      sha256,
+      description,
+      ...
+    }@args:
+    prev.python3.pkgs.buildPythonPackage (
+      {
+        format = "wheel";
+        src = prev.fetchurl { inherit url sha256; };
+        meta = {
+          inherit description;
+          inherit (prev.azure-cli.meta) platforms maintainers;
+          homepage = "https://github.com/Azure/azure-cli-extensions";
+          changelog = "https://github.com/Azure/azure-cli-extensions/blob/main/src/${pname}/HISTORY.rst";
+          license = prev.lib.licenses.mit;
+          sourceProvenance = [ prev.lib.sourceTypes.fromSource ];
+        } // args.meta or { };
+      }
+      // (removeAttrs args [
+        "url"
+        "sha256"
+        "description"
+        "meta"
+      ])
+    );
+
+  confcom = mkAzExtension rec {
+    pname = "confcom";
+    version = "1.0.0";
+    url = "https://azcliprod.blob.core.windows.net/cli-extensions/confcom-${version}-py3-none-any.whl";
```

Comment:
The idea of this PR is to detect AKS changes that break the confcom/genpolicy versions that we base our software on, and thus are likely to affect us?

Reply:
The idea of this PR is to make a test for azures own stuff to detect if we can upgrade our own components or if microsoft broke something. The addition of the `confcom` package is just so that I can generate a good policy for kata since our own `genpolicy` doesn't work on the `kata-cc-isolation` runtime class.

Reply:
My concern here is that this tests a certain fixed configuration, whereas Microsoft considers supported whatever is currently available on AKS and the latest release of confcom. If we wanted to show that AKS is working like MS intended, we would need to use their instructions.

On the other hand, it might be useful to test whether the unpatched sources we build on are compatible with `kata-cc-isolation`, but then I'd rather take `.#msft.genpolicy`.

Reply:
Since this just deploys a container on AKS it uses their upstream config, and the confcom extension will be updated by myself when a new version comes out. Depending on how often we update our `flake.lock` we _may_ fall behind the newest `confcom` version. An option to solve that would be to keep a `confcom_version.txt` and override the package version in nixpkgs upstream with our own version number, which would bypass whole flake updates.